notebooks/default-prep-script.ipynb

```diff
@@ -0,0 +1,417 @@
+   "execution_count": 1,
+   "id": "e3736f91-5695-4beb-8591-227eb444adc0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "import datetime\n",
+    "import math\n",
+    "import numpy\n",
+    "import os\n",
+    "import requests\n",
+    "import stat\n",
+    "import sys\n",
+    "import time\n",
+    "\n",
+    "import sycamore\n",
+    "from sycamore.functions import HuggingFaceTokenizer, TextOverlapChunker\n",
+    "from sycamore.llms import OpenAI, OpenAIModels\n",
+    "from sycamore.transforms.embed import SentenceTransformerEmbedder\n",
+    "from sycamore.transforms.extract_entity import OpenAIEntityExtractor\n",
+    "from sycamore.transforms.merge_elements import GreedyTextElementMerger\n",
+    "from sycamore.transforms.partition import UnstructuredPdfPartitioner, HtmlPartitioner\n",
+    "\n",
+    "index = \"demoindex0\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "6ff7cdd1-4e25-47eb-9a42-eed137346118",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def find_recursive(root, file_type):\n",
+    "    prefix = Path(root)\n",
+    "    type_root = os.path.join(prefix, file_type)\n",
+    "    ret = []\n",
+    "    for path in Path(type_root).rglob(\"*\"):\n",
+    "        ret.append({\"path\": path, \"suffix\": path.relative_to(prefix), \"type\": file_type})\n",
+    "\n",
+    "    return ret\n",
+    "\n",
+    "def find_files(root):\n",
+    "    downloads = root.joinpath(\"downloads\")\n",
+    "    pdf_files = find_recursive(downloads, \"pdf\")\n",
+    "    html_files = find_recursive(downloads, \"html\")\n",
+    "    return pdf_files + html_files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "04cb55d4-3ad7-4003-bc09-6ea1b1ca4432",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/app/work\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100 2797k  100 2797k    0     0  2856k      0 --:--:-- --:--:-- --:--:-- 2854k\n",
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100 62509  100 62509    0     0   140k      0 --:--:-- --:--:-- --:--:--  140k\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "pwd\n",
+    "sudo mkdir -p /app/work/crawl_data/downloads/pdf\n",
+    "sudo mkdir -p /app/work/crawl_data/downloads/html\n",
+    "sudo curl https://sortbenchmark.org/ELSAR2022.pdf -o /app/work/crawl_data/downloads/pdf/elsar.pdf\n",
+    "sudo curl https://sortbenchmark.org -o /app/work/crawl_data/downloads/html/home.html"
+   ]
+  },
+  {
```

Comment:
I think this cell is unneeded, if you want to show the files, I'd just do:
! find /app/work/crawl_data/downloads -print
But I don't think that's helpful, the curl lines either download or fail.